app/ydoc-shared/src/ast/tree.ts

```diff
@@ -1,22 +1,27 @@
+// Declaration-merging is used to implement mixin types in this file.
 /* eslint-disable @typescript-eslint/no-unsafe-declaration-merging */
+
+// Although to the typechecker an empty interface extending a type is equivalent to a type alias resolving to the type,
+// my IDE (JetBrains) resolves type aliases (even if they point to the internal name of an unexported type), but leaves
+// intact references to an interface.
+/* eslint-disable @typescript-eslint/no-empty-object-type */
```

Comment:
This is also true in VSCode. We should honestly just remove that lint rule altogether.